ftp_blueprints/download_file.py

```diff
@@ -197,14 +204,16 @@ def get_client(host, port, username, password):
     specified credentials
     """
     try:
-        client = ftplib.FTP()
+        client = ftplib.FTP(timeout=3600)
         client.connect(host, int(port))
         client.login(username, password)
+        client.set_pasv(True)
+        client.set_debuglevel(0)
```

Comment:
This is in-line with what I was seeing when I was digging into the issue. Not sure if that was the _main_ issue you discovered, but this should help cover some other potential issues.